lib/Backend/GlobOptIntBounds.cpp

```diff
@@ -174,7 +174,14 @@ bool ValueInfo::TryGetInt64ConstantValue(int64 *const intValueRef, const bool is
         int32 int32ValueRef;
         if (TryGetIntConstantValue(&int32ValueRef, false))
         {
-            *intValueRef = (isUnsigned) ? (uint)int32ValueRef : int32ValueRef;
+            if (isUnsigned)
```

Comment:
That's a very subtle bug hahaha
I guess implicit conversion warning would have caught this.
Would `*intValueRef = (isUnsigned) ? (int64)(uint)int32ValueRef : (int64)int32ValueRef;` have also been a valid fix ?